api/src/main/java/org/bahmni/module/hwcinventory/service/impl/EsanjeevaniServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+package org.bahmni.module.hwcinventory.service.impl;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.velocity.runtime.log.Log;
+import org.bahmni.module.hwcinventory.contract.CreatePatientRequest;
+import org.bahmni.module.hwcinventory.contract.LoginRequest;
+import org.bahmni.module.hwcinventory.mapper.EsanjeevaniPatientMapper;
+import org.bahmni.module.hwcinventory.service.EsanjeevaniService;
+import org.bahmni.module.hwcinventory.util.PasswordUtil;
+import org.openmrs.Patient;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.context.Context;
+import org.springframework.beans.factory.annotation.Autowired;
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class EsanjeevaniServiceImpl implements EsanjeevaniService {
+
+    @Autowired
+    PatientService patientService;
+    private Log log;
+    private static final java.util.logging.Logger logger = java.util.logging.Logger.getLogger(EsanjeevaniServiceImpl.class.getName());
+
+    @Override
+    public String getSSOUrl(String ssoLoginResponse) throws Exception {
+        String referenceId = extractReferenceId(ssoLoginResponse);
+        return Context.getAdministrationService().getGlobalProperty("esanjeevani.login.url") + referenceId;
```

Comment:
Should be a constant extracted above (esanjeevani login url)

Reply:
Done